src/Data/List/Properties.agda

```diff
@@ -496,6 +496,16 @@ concat-map {f = f} xss = begin
   foldr (λ ys → map f ys ++_) [] xss         ≡⟨ sym (foldr-fusion (map f) [] (map-++-commute f) xss) ⟩
   map f (concat xss)                         ∎
 
+concat-concat : concat {a} {A} ∘ map concat ≗ concat ∘ concat
+concat-concat [] = refl
+concat-concat ([] ∷ xsss) = concat-concat xsss
+concat-concat (([] ∷ xss) ∷ xsss) = concat-concat (xss ∷ xsss)
+concat-concat (((x ∷ xs) ∷ xss) ∷ xsss) = cong (x ∷_) (concat-concat ((xs ∷ xss) ∷ xsss))
+
```

Comment:
Feels like these last cases could be made cleaner by first proving
`concat-++ : concat (xs ++ ys) ≡ concat xs ++ concat ys`
and then using equational reasoning.